base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -186,46 +203,65 @@ where T: BlockchainBackend
         })
     }
 
+    fn update_metadata(&self, new_height: u64, new_hash: Vec<u8>) -> Result<(), ChainStorageError> {
+        let mut db = self.metadata.write().map_err(|_| {
+            ChainStorageError::AccessError(
+                "Could not obtain write access to blockchain metadata after storing block".into(),
+            )
+        })?;
+        db.height_of_longest_chain = Some(new_height);
+        db.best_block = Some(new_hash);
+        Ok(())
+    }
+
     /// Returns the height of the current longest chain. This method will only fail if there's a fairly serious
     /// synchronisation problem on the database. You can try calling [BlockchainDatabase::try_recover_metadata] in
     /// that case to re-sync the metadata; or else just exit the program.
-    pub fn get_height(&self) -> Result<u64, ChainStorageError> {
-        let metadata = self.get_metadata()?;
+    ///
+    /// If the chain is empty (the genesis block hasn't been added yet), this function returns `None`
+    pub fn get_height(&self) -> Result<Option<u64>, ChainStorageError> {
```

Comment:
How about returning 0 height instead of None? Will save a lot of unwrap kinda code and there are very few times where the chain will be empty

Reply:
Returning 0 was broken. 0 is a valid height since the Genesis block is at height 0